File: Introduction to Robotics Course/Chapter12/Lesson1/Chapter12_Lesson1.py
# publisher_py.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import Float64

class SinePublisher(Node):
    def __init__(self):
        super().__init__('sine_publisher')
        self.pub = self.create_publisher(Float64, 'sine', 10)
        self.k = 0
        self.timer = self.create_timer(0.01, self.step)  # 100 Hz

    def step(self):
        msg = Float64()
        msg.data = float(__import__('math').sin(0.1 * self.k))
        self.pub.publish(msg)
        self.k += 1

def main():
    rclpy.init()
    node = SinePublisher()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()
      

File: Introduction to Robotics Course/Chapter12/Lesson1/Chapter12_Lesson1.java
// PublisherJava.java (conceptual skeleton)
import org.ros.node.AbstractNodeMain;
import org.ros.node.ConnectedNode;
import org.ros.node.topic.Publisher;
import std_msgs.Float64;

public class PublisherJava extends AbstractNodeMain {
  @Override
  public void onStart(final ConnectedNode connectedNode) {
    final Publisher<Float64> pub =
        connectedNode.newPublisher("sine", Float64._TYPE);

    connectedNode.executeCancellableLoop(new org.ros.concurrent.CancellableLoop() {
      private int k = 0;
      @Override
      protected void loop() throws InterruptedException {
        Float64 msg = pub.newMessage();
        msg.setData(Math.sin(0.1 * k));
        pub.publish(msg);
        k++;
        Thread.sleep(10); // 100 Hz
      }
    });
  }

  @Override
  public org.ros.namespace.GraphName getDefaultNodeName() {
    return org.ros.namespace.GraphName.of("sine_publisher_java");
  }
}
      
